docs/excel/custom-functions-dynamic-arrays.md

```diff
@@ -0,0 +1,58 @@
+---
+ms.date: 12/16/2019
+description: Return a dynamic array from your custom function in an Office Excel add-in.
+title: Return a dynamic array from your custom function
+localization_priority: Normal
+---
+
+# Return a dynamic array from your custom function
+
+You can return multiple results from your custom function to make it a dynamic array formula. Multiple results will spill into neighboring cells. You can return results that only spill down, to the right, or both down and to the right.
+
+To create a custom function that is a dynamic array formula, it must return a two-dimensional array of values. The following example shows how to return a dynamic array that spills down.
+
+```javascript
+/**
+ * Get text values that spill down.
+ * @customfunction
+ * @returns {string[][]} A dynamic array with multiple results.
+ */
+function spillDown() {
+  return [['first'], ['second'], ['third']];
+}
+```
+
+The following example shows how to return a dynamic array that spills right. 
+
+```javascript
+/**
+ * Get text values that spill to the right.
+ * @customfunction
+ * @returns {string[][]} A dynamic array with multiple results.
+ */
+function spillRight() {
+  return [['first', 'second', 'third']];
+}
+```
+
+The following example shows how to return a dynamic array that spills both down and right.
+
+```javascript
+/**
+ * Get text values that spill both right and down.
+ * @customfunction
+ * @returns {string[][]} A dynamic array with multiple results.
+ */
+function spillRectangle() {
+  return [
+    ['apples', 1, 'pounds'],
+    ['oranges', 3, 'pounds'],
+    ['pears', 5, 'crates']
+  ];
+}
+```
+
```

Comment:
Support article linked below says that: "Excel has limited support for dynamic arrays between workbooks, and this scenario is only supported when both workbooks are open. If you close the source workbook, any linked dynamic array formulas will return a #REF! error when they are refreshed." I would imagine the same is true for CFs? Also would this be something to call out in the "frequent errors" section of our troubleshooting docs?

Reply:
I think this applies to any formula that links, and therefore any custom function, so this really wouldn't be the place to document #REF! behavior. However, this again is something the end user needs to be aware of. Custom functions are not behaving any differently than the built-in functions when linking (as far as I know.)